### PR TITLE
Backend: Replace round and roundToPrecision with roundTo

### PR DESCRIPTION
**Pull request by CalMWolfs**
## What
Replace round and roundToPrecision with roundTo.

## Changelog Technical Details
+ Removed duplicate rounding functions. - CalMWolfs

### CODE REVIEW

**Review by ThatGravyBoat — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/features/garden/GardenYawAndPitch.kt [hunk 1]
```diff
@@ -43,8 +43,8 @@ object GardenYawAndPitch {
 
         if (!config.showAlways && lastChange.passedSince() > config.timeout.seconds) return
 
-        val yawText = yaw.round(config.yawPrecision).toBigDecimal().toPlainString()
-        val pitchText = pitch.round(config.pitchPrecision).toBigDecimal().toPlainString()
+        val yawText = yaw.roundTo(config.yawPrecision).toBigDecimal().toPlainString()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/garden/GardenYawAndPitch.kt:44]: the change breaks here

File: src/main/java/at/hannibal2/skyhanni/utils/LorenzUtils.kt [hunk 2]
```diff
@@ -98,23 +99,11 @@ object LorenzUtils {
         return originalMessage.stripHypixelMessage()
     }
 
-    fun Double.round(decimals: Int): Double {
-        var multiplier = 1.0
-        repeat(decimals) { multiplier *= 10 }
-        val result = kotlin.math.round(this * multiplier) / multiplier
-        val a = result.toString()
-        val b = toString()
-        return if (a.length > b.length) this else result
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/utils/LorenzUtils.kt:109]: removing this workaround breaks stuff

**Review by hannibal002 — CHANGES_REQUESTED** [review]
<details>
<summary>Breaks on high rounding values</summary>


image
image
image

https://play.kotlinlang.org/#eyJ2ZXJzaW9uIjoiMS45LjI0IiwicGxhdGZvcm0iOiJqYXZhIiwiYXJncyI6IiIsIm5vbmVNYXJrZXJzIjp0cnVlLCJ0aGVtZSI6ImlkZWEiLCJjb2RlIjoiLyoqXG4gKiBZb3UgY2FuIGVkaXQsIHJ1biwgYW5kIHNoYXJlIHRoaXMgY29kZS5cbiAqIHBsYXkua290bGlubGFuZy5vcmdcbiAqL1xuaW1wb3J0IGtvdGxpbi5tYXRoLnBvd1xuaW1wb3J0IGtvdGxpbi5tYXRoLnJvdW5kVG9JbnRcblxuICAgIGZ1biBEb3VibGUucm91bmRUbyhwcmVjaXNpb246IEludCk6IERvdWJsZSB7XG4gICAgICAgIHZhbCBzY2FsZSA9IDEwLjAucG93KHByZWNpc2lvbikudG9JbnQoKVxuICAgICAgICByZXR1cm4gKHRoaXMgKiBzY2FsZSkucm91bmRUb0ludCgpLnRvRG91YmxlKCkgLyBzY2FsZVxuICAgIH1cblxuZnVuIG1haW4oKSB7XG4gICAgdmFsIHJlc3VsdCA9IC0xMTUuMTIzNDU2Ny5yb3VuZFRvKDEwKVxuICAgIHByaW50KHJlc3VsdClcbn0ifQ==

</details>

**Comment by CalMWolfs** [comment]
> Breaks on high rounding values

fixed: https://pl.kotl.in/FBDmugW6X

**Pull request merged**